site/src/components/PortForwardButton/PortForwardButton.tsx

```diff
@@ -0,0 +1,156 @@
+
+  return (
+    <Stack direction="column" spacing={1}>
+      <HelpTooltipText>
+        You can port forward this resource by typing the{" "}
+        <strong>port, workspace name, agent name</strong> and <strong>your username</strong> in the
+        URL like the example below
+      </HelpTooltipText>
+
+      <CodeExample code={urlExample} />
+
+      <HelpTooltipText>Or you can use the following form to open it in a new tab.</HelpTooltipText>
+
+      <Stack direction="row" spacing={1} alignItems="center">
+        <TextField
+          label="Port"
+          type="number"
+          value={port}
+          className={styles.portField}
+          onChange={(e) => {
+            setPort(e.currentTarget.value)
+          }}
+        />
+        <Link
+          underline="none"
+          href={urlExample}
+          target="_blank"
+          rel="noreferrer"
+          className={styles.openUrlButton}
+        >
+          <Button>Open URL</Button>
+        </Link>
+      </Stack>
+
+      <HelpTooltipLinksGroup>
+        <HelpTooltipLink href="https://coder.com/docs/coder-oss/latest/port-forward">
+          Learn more about port forward
+        </HelpTooltipLink>
+      </HelpTooltipLinksGroup>
+    </Stack>
+  )
+}
+
+const DisabledView: React.FC<PortForwardButtonProps> = () => {
+  return (
+    <Stack direction="column" spacing={1}>
+      <HelpTooltipText>
+        <strong>Looks like your deployment does not have port forward enabled.</strong> Check the
+        docs for more info on how to enable it.
+      </HelpTooltipText>
+
+      <HelpTooltipLinksGroup>
+        <HelpTooltipLink href="https://coder.com/docs/coder-oss/latest/networking/port-forwarding">
+          Learn more about port forward
+        </HelpTooltipLink>
+      </HelpTooltipLinksGroup>
+    </Stack>
+  )
+}
+
+export const PortForwardButton: React.FC<PortForwardButtonProps> = (props) => {
+  const { host } = props
+  const anchorRef = useRef<HTMLButtonElement>(null)
+  const [isOpen, setIsOpen] = useState(false)
+  const id = isOpen ? "schedule-popover" : undefined
+  const styles = useStyles()
+
+  const onClose = () => {
+    setIsOpen(false)
+  }
+
+  return (
+    <>
+      <Button
+        startIcon={<OpenInNewOutlined />}
+        size="small"
+        ref={anchorRef}
+        onClick={() => {
+          setIsOpen(true)
+        }}
+      >
+        Port forward
+      </Button>
+      <Popover
+        classes={{ paper: styles.popoverPaper }}
+        id={id}
+        open={isOpen}
+        anchorEl={anchorRef.current}
+        onClose={onClose}
+        anchorOrigin={{
+          vertical: "bottom",
+          horizontal: "left",
+        }}
+        transformOrigin={{
+          vertical: "top",
+          horizontal: "left",
+        }}
+      >
+        <ChooseOne>
+          <Cond condition={host !== ""}>
+            <EnabledView {...props} />
+          </Cond>
+          <Cond condition={host === ""}>
+            <DisabledView {...props} />
```

Comment:
Down the road, there may be deployments that will never want to show this "port forward" dialog to users (e.g. they have security constraints that prevent them from generating a wildcard).

For now, I think it is best for us to keep it (to inform admins they can enable the feature). Just a note we might have to do something similar to #4131 down the road :)

Reply:
Interesting, thanks for letting me know.